### PR TITLE
Added KNN implementation

### PR DESCRIPTION
**Pull request by NitishPradhan26**

### CODE REVIEW

**Comment by longtv2222** [comment]
Hey Nitish, the initial solution is looking good, but I feel like it can be polished more. Something I notice that are:

image

- Can you fix this typo and heading to match mine

image
- I think what we should do here is that we need to have 3 sets: train, validation, and test. And we should report each score for train, validation and test set as well. I've created a function called `report_result` so that we have an unified way to report our result. 
- Why is `n_neighbors` 4 here, I thought that since this is a binary classification problem, it would be better as 2
- In the validation set, we should hyperparameter tuning the PCA component, the range can be 1 to n_features. You can have a look at my work on how to do that, you can also refer [this link](https://www.projectpro.io/recipes/optimize-hyper-parameters-of-decisiontree-model-using-grid-search-in-python#mcetoc_1g1ajorna9) on how to do that. This way the TA can see how we achieve out result
- In addition to PCA, you can hyperparameter tuning KNN `weights` and `algorithm` as well

image

**Comment by NitishPradhan26** [comment]
For the train, test, validate comment: my paper does not implement that

For why neighbours is 4, I tested the accuracy score with multiple neighbours and 4 gave the most accuracy. And the number of classes does not mean we should use the same number of neighbours. From what i read from an article the best way to find the amount of neighbours is to test with different number and choose the number that gives the best results which in this case is 4.
I will try the hyper tuning for kneighbours to see if the results get better

**Comment by NitishPradhan26** [comment]
Tried the hyper tuning for KNN parameter no significant changes
